client/signup/steps/domains/index.jsx

```diff
@@ -343,7 +338,11 @@ class DomainsStep extends Component {
 				},
 				this.getThemeArgs()
 			),
-			Object.assign( { domainItem }, useThemeHeadstartItem )
+			Object.assign(
+				{ domainItem },
+				useThemeHeadstartItem,
+				signupDomainOrigin ? { signupDomainOrigin } : {}
```

Comment:
That makes sense. I updated the PR to set the `signupDomainOrigin` directly within the `handleAddMapping ` so I can avoid prop drilling. This simplifies the PR a lot since I don't need the whole `onConnect` logic